windowsforms/Sudoku/VB/sudoku/core/Game.vb

```diff
@@ -0,0 +1,57 @@
+Public Class Game
+
+    Public Event ShowClues(grid()() As Integer)
+    Public Event ShowSolution(grid()() As Integer)
+
+    Private ReadOnly HRow(8) As List(Of Integer)
+    Private ReadOnly VRow(8) As List(Of Integer)
+    Private ReadOnly ThreeSquare(8) As List(Of Integer)
+    Private ReadOnly grid(8)() As Integer
+
+    Private r As Random
+
+    Public Sub NewGame(rn As Random)
+        Me.r = rn
+        createNewGame()
+    End Sub
```

Comment:
Why this NewGame method needs a Random object to be passed ? In my opinion, it's better to create it internally. Also I'd make the methods PascalCase not camelCase. I also think that the `r` private field can be made ReadOnly.

```
Public Sub NewGame()
    Me.r = New Random()
    CreateNewGame()
End Sub
```

I'm also thinking of why you made this a normal method ? why not a constructor ?

Reply:
Again this is the original code there are a lot of things I would change if I were writing it today.

Reply:
Passing in rn allows testing by using a fixed seed in the Test code but I don't think that was ever implemented.

Reply:
The Random function is actually used in Form.game.ShowClues and Game and that is why it is passed. I there could be 2 and it would not change anything.